crates/sc-proof-of-time/src/source.rs

```diff
@@ -335,24 +335,54 @@ where
         // This will do one of 3 things depending on circumstances:
         // * if block import is ahead of timekeeper and gossip, it will update next slot input
         // * if block import is on a different PoT chain, it will update next slot input to the
-        //   correct fork
+        //   correct fork (reorg)
         // * if block import is on the same PoT chain this will essentially do nothing
-        if let Some(next_slot_input) = self.state.update(
+        match self.state.update(
             best_slot,
             best_proof,
             Some(subspace_digest_items.pot_parameters_change),
         ) {
-            warn!("Proof of time chain reorg happened");
+            PotStateUpdateOutcome::NoChange => {
+                trace!(
+                    %best_slot,
+                    "Block import didn't result in proof of time chain changes",
+                );
+            }
+            PotStateUpdateOutcome::Extension { from, to } => {
+                warn!(
+                    from_next_slot = %from.slot,
+                    to_next_slot = %to.slot,
+                    "Proof of time chain was extended from block import",
+                );
 
-            if self
-                .to_gossip_sender
-                .try_send(ToGossipMessage::NextSlotInput(next_slot_input))
-                .is_err()
-            {
-                debug!(
-                    next_slot = %next_slot_input.slot,
-                    "Gossip is not able to keep-up with slot production (block import)",
+                if self
```

Comment:
Consider introducing a boolean here and move the duplicated snippet out of the match. An alternative would be creating a logging function with `PotStateUpdateOutcome` as an argument. It could trace or warn and the rest of the code could be inside of `if let`. The current duplicated snippet could lead to changes in the single branch.

Reply:
The changes to a single branch might also be desirable. By introducing a boolean here we make code less linear and probably harder to follow. Separate logging function is even worse because it makes things even less linear.

This suggestion seems like a DRY for the purposes of DRY even if it makes code more complex as the result.